### DISCUSSION TITLE
Power consumption and more active use of sleep modes

### DISCUSSION THREAD

**Discussion by cfsorensen** [UtilitechAS/amsreader-firmware#506] [General]
Hi,
I have been playing around with this awesome software for a while, and I love it.
I have tried to power it using the Power Meter, but it only provide approx 4,15V and 75mA, which drains my setup using a ESP32wroom within some days even though I have build backup power with rechargeable batteries, so I wonder how I could make the board sleep more for the batteries to charge more.
I believe that it currently transfer data every 10th second or something similar to that, and to my understanding this wouldn't allow to use the sleep mode, but during night it would be fully fine to only read 1 or 2 times per hour.
Any guidance to enable such a setup would be much appreciated.
Thanks,

**Comment by ArnieO** [comment]
It wakes from auto modem sleep at each payload from the meter. On a Kamstrup the interval between payloads is 10 seconds, on other meters it is even shorter. It never goes lower than modem sleep.

Not really an answer to your question, but if you want to run it on battery for a longer period, try an ESP32 type that consumes less. It will help somewhat, as the ESP32-WROOM is quite power hungry.

**Reply by cfsorensen** [reply]
Hi ArnieO,

Thank you for your reply!

Without having to change the board I find it interesting to try to see if I could make it work with the Wroom. Would it be possible to make the software not wakeup at each payload in a specific timeslot? If so it would be much appreciated to get any directions about where in the code this additional condition should be added.

Thanks,

**Reply by ArnieO** [reply]
Your project sounds technically feasible. If I were you I certainly not insist on using a Wroom, but instead use one of the newer Espressif modules available that consume less such as ESP32-S2 and ESP32-C3. (Our products use ESP32-S2). 

But assuming a world where the old WROOM was the only module available I would do the following:

**Hardware**
Add a large low-ESR (which is an important point) super capacitor after the 3,3V voltage regulator to ensure there is enough juice available when the WROOM is active on Wifi. I think this is the only way to make such a setup stable and reliable. I believe the Wroom draws short bursts of around 200 mA when it is active on Wifi.

**Firmware**
Insert a timed deepsleep command in the main loop, then tune the sleep duration to something that works and suits you, (the longer the better, as battery charging will be a slow process at the available current). I would use the Vcc measurement feature of this firmware to monitor the battery in stead of Vcc (3,3V). You do that be inserting a suitable voltage divider. We use 22k + 33k on our boards to measure 3,3V, you'll have to do some calculations to check if that works also at the max voltage of your rechargeable battery.

Don't expect to be able to do one-click or OTA firmware updates without connecting external power while that is ongoing. 

Good luck with your project!
